### PR TITLE
Show inherited configuration properties in the UI

### PR DESCRIPTION
**Pull request by laurentiumagureanu**
## Description

This feature allows creation of configurations that inherit properties (enabled by default) and override just the necessary ones. 
This is inline with the "inherited properties" pattern used everywhere in AEM.

## Related Issue

CIF-1326

## Motivation and Context

Alignment with the "inherited properties" pattern used everywhere in AEM.

## How Has This Been Tested?

Manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/configuration/servlets/ConfigurationInheritanceServlet.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            if (configContainerPath.startsWith("/conf") &&
+                parentConfigContainerResource != null &&
+                !parentConfigContainerResource.getPath().equals("/conf")) {
+
+                // Set inherited prop true and add a new JSON object to hold inherited properties values
+                config.put("inherited", true);
+                JSONObject inheritedProps = new JSONObject();
+
+                Resource parentConfig;
+                while (parentConfigContainerResource != null && !parentConfigContainerResource.getPath().equals("/conf")) {
```
Comment (herzog31) [inline bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/configuration/servlets/ConfigurationInheritanceServlet.java]: Here you're checking for inheritance within `/conf`. What about `/libs/settings/cloudconfigs/commerce` and a potential overlay in `/apps`?

File: bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/configuration/servlets/ConfigurationInheritanceServlet.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            // Get the container resource
+            Resource configContainerResource = request.getResourceResolver().getResource(configContainerPath);
+
+            if (configContainerResource == null) {
+                response.sendError(500, "Unable to load configuration container resource");
+                return;
+            }
+
+            // Start with a JSON documents that assumes no inheritance
+            JSONObject config = new JSONObject();
```
Comment (herzog31) [inline bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/configuration/servlets/ConfigurationInheritanceServlet.java]: I'm a bit selfish here 😁  Could you please refactor the crawling logic into a separate class so we could also use it for CIF-1523?

**Review by herzog31 — COMMENTED** [review]
Please don't forget about testing :)

**Review by dplaton — CHANGES_REQUESTED** [review]
As @herzog31 pointed out, we should aim to make the "crawling" logic reusable.

**Review by herzog31 — APPROVED** [review]

**Review by dplaton — APPROVED** [review]

**Pull request merged**